src/javaharness/README.md

```diff
@@ -0,0 +1,27 @@
+## Java glue layer for Shell JS and Android 
+
+Code here implements a thin layer that is portable between 
+Android and J2CL->JS so allow developing rapidly without 
+rebuilding Android apps.
+
+Code in arcs/api is platform independent and should have no dependencies on Android API runtime.
+
+Code in arcs/webimpl is platform dependent and expected to be reimplemented in arcs/androidimpl
+
+Native particles will reside and be registered in arcs/nativeparticles
+
+Note there are symbolic links between pipes-shell and
+the outer shells/pipes-shell directory, that means the pipes-shell 
+must be build first.
+
+## Running
```

Comment:
What version of Java is required and how to ensure as part of install/setup/build that the developer has it?

Reply:
Java8 or above. Most MacOS developer devices already have  Java I think, but I guess we can add a script to download it.